File: tools/validate.py
#!/usr/bin/env python3
import json
import sys
from pathlib import Path

from jsonschema import Draft202012Validator

ROOT = Path(__file__).resolve().parents[1]
SCHEMA_PATH = ROOT / "schema" / "agent-definition.schema.json"


def main(path: str) -> None:
    schema = json.loads(SCHEMA_PATH.read_text())
    instance = json.loads(Path(path).read_text())

    validator = Draft202012Validator(schema)
    errors = sorted(validator.iter_errors(instance), key=lambda e: e.path)

    if errors:
        print(f"❌ {path} is NOT valid:")
        for err in errors:
            loc = ".".join(str(x) for x in err.path) or "<root>"
            print(f"  - {loc}: {err.message}")
        sys.exit(1)
    else:
        print(f"✅ {path} is valid against agent-definition.schema.json")


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python tools/validate.py <path-to-agent-json>")
        sys.exit(1)
    main(sys.argv[1])

File: tools/validate.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");
const Ajv = require("ajv");
const addFormats = require("ajv-formats");

const ROOT = path.resolve(__dirname, "..");
const schemaPath = path.join(ROOT, "schema", "agent-definition.schema.json");

if (process.argv.length !== 3) {
  console.error("Usage: node tools/validate.js <path-to-agent-json>");
  process.exit(1);
}

const agentPath = path.resolve(process.argv[2]);
const schema = JSON.parse(fs.readFileSync(schemaPath, "utf8"));
const instance = JSON.parse(fs.readFileSync(agentPath, "utf8"));

const ajv = new Ajv({ allErrors: true, strict: false });
addFormats(ajv);
const validate = ajv.compile(schema);
const valid = validate(instance);

if (!valid) {
  console.error(`❌ ${agentPath} is NOT valid:`);
  console.error(validate.errors);
  process.exit(1);
} else {
  console.log(`✅ ${agentPath} is valid against agent-definition.schema.json`);
}
